lesson02.md

```diff
@@ -0,0 +1,399 @@
+<!-- .slide: id="lesson2" -->
+
+# Basic Frontend - Spring 2021
+
+Lesson 2, Thursday, 2021-03-04
+
+(Remember to install Visual Studio Code if you haven't already)
+
+---
+
+### Recap
+
+What did we talk about last lesson?
+
+---
+
+### Recap: data types
+
+Number:
+```js
+42
+-3.1415
+```
+
+String:
+```js
+"John"
+'Berlin'
+```
+
+Boolean:
+```js
+true
+false
+```
+
+---
+
+### Recap operators
+
+| Group | Operators | Example |
+| ----- | --------- | ------- |
+| Numerical Operators | `+ - * / **` | `5 + 4 * 3`<br/>`7 / 2 - 2`<br/>`"Hello" + " World"` |
+| Comparison Operators | `=== !== < > <= >=` | `30 !== 25`<br/>`20 >= 18` |
+
+---
+
+### Try it out
+
+Now, try to enter some data to the JavaScript console, but think of the data type first:
+
+* Your name
+* Your shoe size
+* Wether you think it's ok to put pineapple on pizza
+
+---
+
+### Try it out: Solution
+
+```js
+"Carlo"
+45
+false
+```
+
+---
+
+### Quiz time
+
+The `typeof` operator in JavaScript outputs the data type of a value. Can you guess the output?
+
+```js
+typeof 42;
+```
+
+```js
+"number"
+```
+<!-- .element: class="fragment" -->
+
+```js
+typeof "hello";
+```
+
+```js
+"string"
+```
+<!-- .element: class="fragment" -->
+
+```js
+typeof true;
+```
+```js
+"boolean"
+```
+<!-- .element: class="fragment" -->
+
+---
+
+### Quiz time
+
+```js
+typeof (1 + 1);
+```
+
+```js
+"number"
+```
+<!-- .element: class="fragment" -->
+
+```js
+typeof ("1" + "1");
+```
+
+```js
+"string"
+```
+<!-- .element: class="fragment" -->
+
+```js
+typeof ("1 + 1");
+```
+```js
+"string"
+```
+<!-- .element: class="fragment" -->
+
+---
+
+### Quiz time
+
+```js
+typeof ("Harald" === "Abdullah");
+```
+
+```js
+"boolean"
+```
+<!-- .element: class="fragment" -->
+
+```js
+typeof (4 < 2);
+```
+
+```js
+"boolean"
+```
+<!-- .element: class="fragment" -->
+
+```js
+typeof typeof 42;
+```
+```js
+"string"
+```
+<!-- .element: class="fragment" -->
+
+---
+
+### Data Type: Undefined
+
+*  **Undefined** has only one value: `undefined`
+* it is just `undefined`
+* it means there is no value here
+
+---
+
+<!-- .slide: id="variables" -->
+
+# Variables
+
+---
+
+### Variables
+
+But with all these examples, we are just talking about values.
+
+What if I wanted to save some values and use them later?
+
+we use `Variables`!
+
+---
+
+### Variable
+
+A pointer to value.
+
```

Comment:
Ask if they ever used an excel sheet to enter some values. What type of values were those?
The numbers and text you entered into cells in excel sheet are the same values  in JS. In Excel you refer to these cells using row and column number, like A1, C5.
You can think of computer's memory as a giant excel sheet. So to put and retrieve any value we want to use for calculation, we need to refer to those cells. So instead of using row and column numbers like in excel, we use names to refer to those cells. This name is called the variable.